lib/kitchen/element_base.rb

```diff
@@ -799,6 +799,31 @@ def as_enumerator
       enumerator_class.new(search_query: search_query_that_found_me) { |block| block.yield(self) }
     end
 
+    def rex_link
+      raise 'Cannot create rex link to an element without an ID' unless id
+
+      element_with_ancestors = document.book.chapters.search_with(
+        Kitchen::PageElementEnumerator, Kitchen::CompositePageElementEnumerator
+      ).search("##{id}").first
+
+      book_slug = document.search('span[data-type="slug"]').first[:'data-value']
+      chapter_count = element_with_ancestors.ancestor(:chapter).count_in(:book)
+      page_string = ''
+      page_title = ''
+      page = element_with_ancestors.ancestor(:page) if element_with_ancestors.has_ancestor?(:page)
+      if page && page&.is_introduction?
+        page_title = page.first('[data-type="document-title"]').text.strip.downcase
+      elsif page
+        page_string = "#{page.count_in(:chapter) - 1}-"
```

Comment:
Can I ask why in the section page case such string was used - `page_in_chapter -1`. I was wondering just out of curiosity why it is always 1and not the numbering we use for sections, like `chapter_number.page_in_chapter_number`.